lambdas/address/src/main/java/uk/gov/di/ipv/cri/address/api/handler/AddressDateLinker.java

```diff
@@ -0,0 +1,102 @@
+package uk.gov.di.ipv.cri.address.api.handler;
+
+import uk.gov.di.ipv.cri.address.library.exception.AddressProcessingException;
+import uk.gov.di.ipv.cri.common.library.annotations.ExcludeFromGeneratedCoverageReport;
+import uk.gov.di.ipv.cri.common.library.persistence.item.CanonicalAddress;
+
+import java.time.LocalDate;
+import java.util.List;
+
+public class AddressDateLinker {
+
+    public static final String ERROR_NO_ADDRESSES = "No Addresses.";
+    public static final String ERROR_SINGLE_ADDRESS_NOT_CURRENT =
+            "A single address found but is not a CURRENT address.";
+    public static final String ERROR_TOO_MANY_ADDRESSES =
+            "Too many Addresses for AddressDateLinker to process.";
+
+    public static final String ERROR_COULD_NOT_DETERMINE_CURRENT_ADDRESS =
+            "Two addresses found but could not determine which address is a CURRENT addressType.";
+
+    public static final String ERROR_ADDRESS_LINKER_NOT_NEEDED =
+            "Address Linker found PREVIOUS address but validUntil was already set - automatic date linking failed.";
+
+    @ExcludeFromGeneratedCoverageReport
+    private AddressDateLinker() {
+        throw new IllegalStateException("Static Utility class");
+    }
+
+    public static void linkCurrentAndPreviousAddressesDates(List<CanonicalAddress> addresses)
+            throws AddressProcessingException {
+
+        switch (addresses.size()) {
+            case 0:
+                throw new AddressProcessingException(ERROR_NO_ADDRESSES);
```

Comment:
if there aren't any addresses this method should simply take no action.
Let's remove the responsibility of throwing an exception when no addresses are present from this method

Reply:
Done, the unit test for this is now inverted and checks for no exceptions with 0 addresses.